### PR TITLE
a not-so-rouge-pull-request

### PR DESCRIPTION
**Pull request by kbilsback**

### CODE REVIEW

File: r_markdown/load_save_data.Rmd [hunk 1]
```diff
@@ -211,6 +211,43 @@ Reference for test id, stove and fuel types.
                              "Liquefied Petroleum Gas")))
 ```
 
+* Define stove fuel combination category levels
+
+```{r, echo=TRUE, warning=FALSE, message=FALSE, results='markup', cache=FALSE}
+  samples <- dplyr::mutate(samples, stove_fuel = as.factor(paste(stove, ":", fuel))) %>%
```
Comment (ngood13) [inline r_markdown/load_save_data.Rmd]: Kelsey!!!!!!!!!!!!!!!!!!!!!!!!!! Stop making me cry!

File: r_markdown/load_save_data.Rmd [hunk 2]
```diff
@@ -211,6 +211,43 @@ Reference for test id, stove and fuel types.
...
+             dplyr::mutate(stove_fuel = factor(stove_fuel,
+                           levels = c("Three Stone Fire (Artisan) : Douglas Fir",
```
Comment (ngood13) [inline r_markdown/load_save_data.Rmd]: Two variables in one variable! They haven't invented an appropriate emoji for this situation yet!

File: r_markdown/load_save_data.Rmd [hunk 3]
```diff
@@ -211,6 +211,43 @@ Reference for test id, stove and fuel types.
...
+                             "Three Stone Fire (Artisan) : Eucalyptus",
+                             "Clay Ring (Artisan) : Douglas Fir",
+                             "Clay Ring (Artisan) : Oak",
+                             "Clay Ring (Artisan) : Eucalyptus",
+                             "Built-in Justa (Envirofit, HM5000) : Douglas Fir",
+                             "Built-in Justa (Envirofit, HM5000) : Oak",
+                             "Built-in Justa (Envirofit, HM5000) : Eucalyptus",
+                             "Rocket Elbow (Envirofit, G3300) : Douglas Fir",
+                             "Rocket Elbow (Envirofit, G3300) : Oak",
+                             "Rocket Elbow (Envirofit, G3300) : Eucalyptus",
```
Comment (ngood13) [inline r_markdown/load_save_data.Rmd]: What's going on here? Could you have done the formatting in the paste statement you used to create the variable?

File: r_markdown/load_save_data.Rmd [hunk 4]
```diff
@@ -234,7 +271,7 @@ Reference for test id, stove and fuel types.
 ```{r, echo=TRUE, warning=FALSE, message=FALSE, results='markup', cache=FALSE}
   samples <- dplyr::mutate(samples,
                            fuelcat = factor(fuelcat,
-                                            levels(fuelcat)[c(3,2,4,5,1)])) %>%
```
Comment (ngood13) [inline r_markdown/load_save_data.Rmd]: I'm sure this is fine, but a little comment to explain the step would be nice.

File: r_markdown/load_save_data.Rmd [hunk 5]
```diff
@@ -234,7 +271,7 @@ Reference for test id, stove and fuel types.
...
+                                            levels(fuelcat)[c(1,2,3,4,5)])) %>%
              dplyr::mutate(stovecat =
                            fct_collapse(stove,
```
Comment (ngood13) [inline r_markdown/load_save_data.Rmd]: use forcats:: (this is Kristen's original code, so note to all)

File: r_markdown/load_save_data.Rmd [hunk 6]
```diff
@@ -250,7 +287,7 @@ Reference for test id, stove and fuel types.
                              advanced = c("Pressurized LPG (WokSmith, WS-C1-25K)",
                                           "Wick Kerosene (Cook 'n Lite, 82)", 
                                           "Pressurized Kerosene (Butterfly, 2412)"))) %>%
-            dplyr::mutate(stovecat = factor(stovecat,levels(stovecat)[c(3,2,4,5,1)]))
```
Comment (ngood13) [inline r_markdown/load_save_data.Rmd]: can we used forcats rather than base for this (and other) factor operations?

Comment (ngood13) [inline r_markdown/load_save_data.Rmd]: spaces!

File: r_markdown/load_save_data.Rmd [hunk 7]
```diff
@@ -250,7 +287,7 @@ Reference for test id, stove and fuel types.
...
+            dplyr::mutate(stovecat = factor(stovecat,levels(stovecat)[c(1,2,3,4,5)]))
```
Comment (ngood13) [inline r_markdown/load_save_data.Rmd]: spaces

File: r_markdown/load_save_data.Rmd [hunk 8]
```diff
@@ -211,6 +211,43 @@ Reference for test id, stove and fuel types.
                              "Liquefied Petroleum Gas")))
 ```
```
Comment (ngood13) [inline r_markdown/load_save_data.Rmd:216]: Sneaky pull request at 17:45 on a Friday?

File: r_markdown/load_save_data.Rmd [hunk 9]
```diff
@@ -250,7 +287,7 @@ Reference for test id, stove and fuel types.
                              advanced = c("Pressurized LPG (WokSmith, WS-C1-25K)",
                                           "Wick Kerosene (Cook 'n Lite, 82)", 
                                           "Pressurized Kerosene (Butterfly, 2412)"))) %>%
-            dplyr::mutate(stovecat = factor(stovecat,levels(stovecat)[c(3,2,4,5,1)]))
+            dplyr::mutate(stovecat = factor(stovecat,levels(stovecat)[c(1,2,3,4,5)]))
 ```
 
 * Save samples file
```
Comment (ngood13) [inline r_markdown/load_save_data.Rmd]: this is a well focused on topic pull A++

**Review by ngood13 — COMMENTED** [review]
Comments added

File: r_markdown/load_save_data.Rmd (hunk 1 above)
Comment (kbilsback) [reply r_markdown/load_save_data.Rmd]: :P

File: r_markdown/load_save_data.Rmd (hunk 7 above)
Comment (kbilsback) [reply r_markdown/load_save_data.Rmd]: that's your code btw :)

File: r_markdown/load_save_data.Rmd (hunk 1 above)
Comment (kbilsback) [reply r_markdown/load_save_data.Rmd]: wait I don't get it my spaces look fine!

**Comment by kbilsback** [comment]
@ngood13 changes made

File: r_markdown/load_save_data.Rmd [hunk 10]
```diff
@@ -9,6 +9,7 @@ output:
 
 ```{r, echo=FALSE, warning=FALSE, message=FALSE, results='markup', cache=FALSE}
   library(tidyverse)
+  library(forcats)
```
Comment (ngood13) [inline r_markdown/load_save_data.Rmd:12]: nice

File: r_markdown/load_save_data.Rmd [hunk 11]
```diff
@@ -178,79 +179,91 @@ Reference for test id, stove and fuel types.
...
-                               advanced = c("Kerosene",
-                                            "Liquefied Petroleum Gas")))
+                           stove_fuel = forcats::as_factor(paste(stove, ":", fuel))) %>%
+                           dplyr::mutate(stove_fuel = 
+                           forcats::lvls_reorder(stove_fuel,
+                                                 c(28, 30, 29, 7, 9, 8,
+                                                   1, 3, 2, 25, 27, 26,
+                                                   10, 12, 11, 6, 5, 4,
+                                                   21, 20, 19, 18, 17, 14,
+                                                   13, 16, 15, 31, 23, 24, 22))) # set logical factor order
```
Comment (ngood13) [inline r_markdown/load_save_data.Rmd:228]: i feel like you should be able to do this once and then when you make the additional categories (like this one and the broader ones) you won't need to reorder?

File: r_markdown/load_save_data.Rmd [hunk 12]
```diff
@@ -178,79 +179,91 @@ Reference for test id, stove and fuel types.
...
-                                        "Eucalyptus",
-                                        "Douglas Fir"),
-                               charcoal = c("Coconut Charcoal",
-                                            "Lump Hardwood (Med.)", 
-                                            "Lump Hardwood (Sm.)"),
-                               pellets = c("Eucalyptus Pellets",
-                                           "West Slope Pellets"),
-                               advanced = c("Kerosene",
-                                            "Liquefied Petroleum Gas")))
+                           stove_fuel = forcats::as_factor(paste(stove, ":", fuel))) %>%
```
Comment (ngood13) [inline r_markdown/load_save_data.Rmd:221]: umm, apparently special KB can get away with anything

**Review by ngood13 — APPROVED** [review]
need to add "no taking advantage of Nick's weakness's" to the style guide. Perhaps we could compromise and calculate the combo's in the metadata rmd? But for now you win :-(

**Pull request merged**